File: src/main/java/net/volcano/jdacommands/model/command/annotations/CommandMethod.java
package net.volcano.jdacommands.model.command.annotations;

import org.springframework.core.annotation.AliasFor;

import java.lang.annotation.*;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface CommandMethod {
	
	@AliasFor("path")
	String[] value() default {""};
	
	@AliasFor("value")
	String[] path() default {""};
	
	String[] permissions() default {};
	
	net.volcano.jdacommands.model.command.Command.Source source() default net.volcano.jdacommands.model.command.Command.Source.BOTH;
	
	boolean sensitive() default false;
	
	boolean global() default false;
	
}

File: src/main/java/net/volcano/jdacommands/model/command/annotations/CommandMethod.kt
package net.volcano.jdacommands.model.command.annotations

import net.volcano.jdacommands.model.command.Source

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class CommandMethod(

	vararg val value: String = [""],

	val path: Array<String> = [""],

	val permissions: String = "",

	val source: Source = Source.DEFAULT,

	val sensitive: Boolean = false,

	val global: Boolean = false

)